daxa/mission/erosita.py

```diff
@@ -0,0 +1,703 @@
+                                "CRAB_III": "CRAB_3", "CRAB_IV": "CRAB_4", "CRAB": "", 
+                                "47_TUC": "47_TUC__NGC_04_", "TGUH2213P1": "TGUH2213P1__DARK_CLOUD_",
+                                "A3391": "A3391_A3395", "A3395": "A3391_A3395"}
+        
+        # Replacing the possible alternative names people could have inputted with the equivalent DAXA friendly formatted one
+        for alt_field in poss_alt_field_names:
+            # In case they just put in crab, again sorry this is so digustingly hard coded
+            if alt_field == "CRAB" and alt_field in fields:
+                # Replacing "CRAB" in the list with the way the links are written in the CalPV_info DataFrame
+                i = fields.index("CRAB")
+                fields[i:i+2] = "CRAB_1", "CRAB_2", "CRAB_3", "CRAB_4"
+            elif alt_field in fields:
+                # Doing this for all the other alternative field name
+                i = fields.index(alt_field)
+                fields[i] = poss_alt_field_names[alt_field]
+            else:
+                bad_fields = [f for f in fields if f not in poss_alt_field_names and f not in self._miss_poss_fields and f not in self._miss_poss_field_types]
+                if len(bad_fields) != 0:
+                    raise ValueError("Some field names or field types {bf} are not associated with this mission, please"
+                            " choose from the following fields; {gf} or field types; {gft}".format(
+                            bf=",".join(bad_fields), gf=",".join(self._miss_poss_fields), gft=",".join(self._miss_poss_field_types)))
+        
+        if all(field in self._miss_poss_field_types for field in fields):
+            # Checking if the fields were given as a field type 
+            # Then collecting all the fields associated with that field type/s
+            updated_fields = CalPV_info.loc[CalPV_info["Field_Type"].isin(fields), "Field_Name"].tolist()
+        elif all(field in self._miss_poss_fields for field in fields):
+            # Checking if the field names are valid
+            updated_fields = fields
+        else:
+            # Checking if any of the fields entered are not valid field names or types
+            field_test = [field in self._miss_poss_field_types or self._miss_poss_fields for field in fields]
+            if not all(field_test):
+                bad_fields = np.array(input_fields)[~np.array(field_test)]
+                raise ValueError("Some field names or field types {bf} are not associated with this mission, please"
+                        "choose from the following fields; {gf} or field types; {gft}".format(
+                        bf=",".join(bad_fields), gf=",".join(self._miss_poss_fields), gft=",".join(self._miss_poss_field_types)))
+            
+            else:
+                # If its got to this stage that means the field where input as a mix of field types and field names
```

Comment:
Ah! After the first comment on this function I went and changed it to work by a case by case basis so that was nice we had the same thinking! It is MUCH easier to read now as well :)